experiments/memcached-cat/main.go

```diff
@@ -196,55 +166,53 @@ func main() {
 					beIsolation := isolation.Rdtset{Mask: beCacheMask, CPURange: beCoresRange}
 					logrus.Debugf("HP isolation: %+v, BE isolation: %+v", hpIsolation, beIsolation)
 
-					// Create executors with cleanup function.
-					hpExecutor, err := sensitivity.CreateKubernetesHpExecutor(hpIsolation)
-					errutil.CheckWithContext(err, "Cannot create executors")
-					beExecutorFactory := sensitivity.DefaultKubernetesBEExecutorFactory
+					workloadFactory := sensitivity.NewWorkloadFactoryWithIsolation(
+						sensitivity.NewExecutorFactory(),
+						hpIsolation,
+						beIsolation,
+						beIsolation,
+					)
 
-					// Clean any RDT assignments from previous phases.
-					pqosOutput, err := isolation.CleanRDTAssingments()
-					logrus.Debugf("pqos -R has been run and produced following output: %q", pqosOutput)
-					errutil.CheckWithContext(err, "pqos -R failed")
+					phaseName := fmt.Sprintf("Aggressor %s (at %d QPS) - BE LLC %b", aggressorName, qps, beCacheMask)
 
-					// Create BE workloads.
-					beLauncher := createLauncherSessionPair(aggressorName, beIsolation, beIsolation, beExecutorFactory)
+					// Building snap workload tags.
+					snapTags := make(map[string]interface{})
+					snapTags[experiment.ExperimentKey] = uid
+					snapTags[experiment.PhaseKey] = phaseName
+					snapTags[experiment.AggressorNameKey] = aggressorName
+					snapTags[experiment.LoadPointQPSKey] = qps
+					snapTags["be_l3_cache_ways"] = beCacheWays
+					snapTags["be_number_of_cores"] = BECPUsCount
+					snapTags["be_cores_range"] = beCoresRange
+					snapTags["hp_cores_range"] = hpCoresRange
 
 					// Create HP workload.
-					memcachedConfig := memcached.DefaultMemcachedConfig()
-					hpLauncher := executor.ServiceLauncher{Launcher: memcached.New(hpExecutor, memcachedConfig)}
+					hpLauncher, err := workloadFactory.BuildDefaultHighPriorityLauncher(sensitivity.Memcached, snapTags)
+					errutil.CheckWithContext(err, "Cannot create Memcached")
```

Comment:
@skonefal I would appreciate having more verbose context information in other calls to `errutil.CheckWithContext()` that you introduced :)